CFC_DataCollector/recommender/recommendation_pipeline.py

```diff
@@ -14,26 +15,24 @@ def retrieve_alternative_trips(self, trip_list):
         return []
 
     def get_selected_user_utility_model(self, user_id):
-        return []
+        return UserUtilityModel.find_from_db(user_id)
 
     def recommend_trips(self, trip_id, utility_model):
         return []
```

Comment:
Ah, the solution for that was not to not use scikit-learn logistic regression, but implement the maximum likelihood optimization manually. In that, I think that you optimize for each [trip,{alternatives}] pair and then multiply the probabilities together (see conversion of eqn 3 to eqn 5 in the Vij/Shankari paper).

Vij sent me his code to do this, and I can send it to you.

But I suspect that we will get similar results by using scikit-learn logistic regression for each trip,{alternatives} set and then multiplying the probabilities together, or at least, good enough results for the first pass. scikit-learn gives you the probabilities.
http://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LogisticRegression.html#sklearn.linear_model.LogisticRegression.predict_proba

There is also a package that Sid is learning in his discrete choice class that he can show you how to use if you want something more complex.